tests/test_sudoku.py

```diff
@@ -44,12 +46,14 @@ def test_exclude_equal_alternatives():
         {7, 8, 9},
         {8, 9, 7},
         {9, 7, 8}
-    ]  
+    ]
 
-    grid_view = [[_cell_from_alternatives(EEA_1ST_ROW_ALTERNATIVES[i]) if j == 0 else Cell() for i in range(9)] for j in range(9)]
+    grid_view = [[_cell_from_alternatives(EEA_1ST_ROW_ALTERNATIVES[i]) if (
+        j == 0) else Cell() for i in range(9)] for j in range(9)]
     Sudoku._Sudoku__exclude_equal_alternatives(grid_view)
-    assert [cell._Cell__alternatives for cell in grid_view[0]] == EEA_1ST_ROW_ALTERNATIVES_SOLUTION # проверяем альтернативы в первой строке
-    assert all([grid_view[i][j]._Cell__alternatives == {1, 2, 3, 4, 5, 6, 7, 8, 9} for j in range(9)] for i in range(1,9)) # проверяем, что альтернативы в других строках не изменлись
+    assert [cell._Cell__alternatives for cell in grid_view[0]] == EEA_1ST_ROW_ALTERNATIVES_SOLUTION
+    assert all([grid_view[i][j]._Cell__alternatives == {1, 2,
+           3, 4, 5, 6, 7, 8, 9} for j in range(9)] for i in range(1, 9))
```

Comment:
Здесь проверяется что-то не то. Проверяется, что вложенные списки не пустые. А нужно проверять равенства. Похоже, что скобки `[...]` лишние.

```suggestion
    assert all(
        grid_view[i][j]._Cell__alternatives == {1, 2,3, 4, 5, 6, 7, 8, 9} 
        for j in range(9) for i in range(1, 9)
    )
```